miditok/constants.py

```diff
@@ -10,16 +10,20 @@
 ADDITIONAL_TOKENS = {'Chord': False,
                      'Rest': False,
                      'Tempo': False,
+                     'TimeSignature': False,
                      'Program': False,
                      # rest params
                      'rest_range': (2, 8),  # (/min_rest, max_rest_in_BEAT), first divides a whole note/rest
                      # tempo params
                      'nb_tempos': 32,  # nb of tempo bins for additional tempo tokens, quantized like velocities
-                     'tempo_range': (40, 250)}  # (min_tempo, max_tempo)
+                     'tempo_range': (40, 250),  # (min_tempo, max_tempo)
+                     # time signature params
+                     'time_signature_range': (8, 2)}  # (max_beat_res, max_bar_length_in_NOTE)
```

Comment:
I had some confronting design decisions for the configuration of time signature tokens. Maybe there is a more convenient way to represent it. For now, the inspiration is taken from MusicBERT. 

1.  `max_beat_res` is the maximum supported beat resolution in time signature vocabulary, a power of 2 (e.g., 8 implies supported time signatures x/1, x/2, x/4, x/8)
2. `max_bar_length_in_NOTE` defines the maximum number of notes and beats in a bar (e.g., 2 implies supported time signatures 1/y, 2/y, ..., 2y/y)

Reply:
When creating time signatures, why ranging the numerator from 0 to `time_signature_range[0] * time_signature_range[1]` ?
Do you know why the author of MusicBERT did so ?

Why not setting `time_signature_range` as `((min_num, max_num), (min_denum, max_denum))` and do:
```
assert math.log2(min_denum).is_integer() and math.log2(max_denum).is_integer(), \
             f'The beat resolution in time signature must be a power of 2'

while denum <= max_denum:
    for num in range(min_num, max_num):
        time_signatures.append((num, 2 ** denum))
    denum *= 2
```
This would leave the user the capability to exclude more easily MIDI files with TS outside these ranges when tokenizing aa dataset

Reply:
@Natooz honestly, I have no idea behind the decisions of MusicBERT authors. 

The assumption "the duration of a bar is no more than two whole" looks fine to cover time signatures like 2/1, 4/2. However, it makes the time signature list very exhaustive and unnatural. For `max_beat_res = 64` they fill vocabulary with probably completely non-existent time signatures like 23/64 or 111/64. I don't like it, but I left it like this as there are no other encodings and implementations with time signature tokens. Correct me if I am mistaken.

The time signature range `((min_num, max_num), (min_denum, max_denum))` is more convenient for users and might reduce the overall time signature list but also create non-existent time signatures. Cases where `max_num` is large and `min_denum` is small (e.g., 9/2, 13/4, etc.).

Maybe different `(min_num, max_num)` for different denominators would fit better? Something like:
```[python]
time_signature_range: {denum_1: (min_num_1, max_num_1), ..., denum_k: (min_num_k, max_num_k)]
```

Personally, I would prefer to provide a list of time signatures present in the dataset to create the vocabulary. But the list might be long enough.